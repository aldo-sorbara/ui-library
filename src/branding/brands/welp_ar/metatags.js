export const presto = {
  title: 'Presto Hoy Prestamos Online',
  description: 'Préstamo confiable, rápido y seguro',
  keywords: 'Préstamo, Crédito, Minicrédito, Presto Hoy, Efectivo, Cash'
};
export const posta = {
  title: 'Créditos Posta',
  description: 'Créditos Posta',
  keywords: 'Préstamo, Crédito, Posta, Efectivo, Cash'
};
export const welp = {
  title: 'W E L P',
  description: 'WELP',
  keywords: 'Préstamo, Crédito, Welp, Efectivo, Cash'
};
export default {
  title: 'Onboarding Title',
  description: 'Onboarding Description',
  keywords: 'Onboarding, React, Redux, Styled'
};
