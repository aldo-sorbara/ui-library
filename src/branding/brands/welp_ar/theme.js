import alerts from './jsons/alerts.json';
import assetsPaths from './jsons/assets.json';
import breakPoints from '../common/jsons/breakPoints.json';
import font from './jsons/font.json';
import palette from './jsons/palette.json';
import reset from './jsons/reset.json';
import spacer from './jsons/spacer.json';
import typography from './jsons/typography.json';
import zIndex from '../common/jsons/zIndex.json';

import buttons from './jsons/buttons.json';
import radioButton from './jsons/radioButton.json';
import list from './jsons/list.json';
import containers from './jsons/containers.json';
import modals from './jsons/modals.json';

export default {
  name: 'welp_ar',
  alerts,
  assetsPaths,
  breakPoints,
  buttons,
  list,
  containers,
  emSize: '16px',
  font,
  modals,
  palette,
  radioButton,
  reset,
  spacer,
  typography,
  zIndex
};
