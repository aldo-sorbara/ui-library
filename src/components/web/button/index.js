export { Button, ButtonContainer } from './Button';
