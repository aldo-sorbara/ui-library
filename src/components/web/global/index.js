export {
  Container,
  Wrapper,
  Headline,
  Description,
  Legales,
  Label,
  LabelLarge,
  SecondHeadline,
  ContainerDesktop,
  Desktop,
  Onboarding,
  Terms,
  H1,
  SectionDescription,
  UrlLink,
  Timer,
  DesktopText,
  Bind,
  Logo,
  LogoMobile,
  TextPromotion
} from './Global';
