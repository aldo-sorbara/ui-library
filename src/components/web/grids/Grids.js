import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const mediaQueries = {
  xs: '576px',
  sm: '768px',
  md: '992px'
};

const StyledContainer = styled.div`
  align-content: unset;
  display: flex;
  flex-wrap: 'no-wrap';
  margin: 0 auto;
  max-width: ${props => props.size}%;
  width: 100%;
`;

export const Container = props => (
  <StyledContainer {...props}>{props.children}</StyledContainer>
);

Container.propTypes = {
  size: PropTypes.number
};

Container.defaultProps = {
  size: 100
};

const StyledGrid = styled.div`
  align-items: unset;
  box-sizing: border-box;
  display: inline-block;
  justify-content: unset;
  order: unset;
  padding: 0;
  vertical-align: top;
  width: ${props => (props.lg === 0 ? 'none' : `${props.lg}`)}%;

  @media (max-width: ${props => props.theme.breakPoints.md}) {
    width: ${props => (props.md === 0 ? 'none' : `${props.md}`)}%;
  }

  @media (max-width: ${props => props.theme.breakPoints.sm}) {
    width: ${props => (props.sm === 0 ? 'none' : `${props.sm}`)}%;
  }

  @media (max-width: ${props => props.theme.breakPoints.xs}) {
    width: ${props => (props.xs === 0 ? 'none' : `${props.xs}`)}%;
  }
`;

StyledGrid.defaultProps = {
  theme: {
    breakPoints: mediaQueries
  }
};

export const Grid = props => (
  <StyledGrid {...props}>{props.children}</StyledGrid>
);

Grid.propTypes = {
  xs: PropTypes.number,
  sm: PropTypes.number,
  md: PropTypes.number,
  lg: PropTypes.number
};

Grid.defaultProps = {
  xs: 0,
  sm: 0,
  md: 0,
  lg: 0
};
