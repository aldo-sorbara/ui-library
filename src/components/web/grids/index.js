export { Grid, Container as GridContainer } from './Grids';
