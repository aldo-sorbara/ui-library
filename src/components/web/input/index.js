export { default, InputLabel, InputDescription } from './Input';
