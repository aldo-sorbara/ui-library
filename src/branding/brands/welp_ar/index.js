import wording from './jsons/wording.json';
import theme from './theme';
import config from './jsons/config.json';
import feedback from './jsons/feedback.json';
import assets from './jsons/assets.json';
import palette from './jsons/palette.json';

export default {
  wording: { onboarding: wording },
  feedback: { onboarding: feedback },
  theme,
  config: { onboarding: config },
  typography: theme.font.fontfamily,
  assets: { onboarding: assets },
  palette
};
