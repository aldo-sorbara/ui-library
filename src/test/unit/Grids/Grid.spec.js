import React from 'react';
import { Grid, GridContainer } from '../../../components/web/grids';
import renderer from 'react-test-renderer';
import { mountWithTheme } from '../../../../utils/helper';
import welp_ar from '../../../branding/brands/welp_ar';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

describe('Grid component', () => {
  it('should render snapshot (only grid)', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Grid lg={25} md={25} sm={25} xs={25}>
          This is a Grid component
        </Grid>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render snapshot (with container)', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <GridContainer>
          <Grid lg={75} md={75} sm={75} xs={75}>
            This is a Grid component with container
          </Grid>
        </GridContainer>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render a default grid', () => {
    const wrapper = mountWithTheme(
      <Grid> This is a default Grid component </Grid>,
      welp_ar.theme
    );
    expect(wrapper.props().lg).toBe(0);
    expect(wrapper.props().md).toBe(0);
    expect(wrapper.props().sm).toBe(0);
    expect(wrapper.props().xs).toBe(0);
  });
});
