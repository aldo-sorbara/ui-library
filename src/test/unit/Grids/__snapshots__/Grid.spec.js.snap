// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component should render snapshot (only grid) 1`] = `
.c0 {
  -webkit-align-items: unset;
  -webkit-box-align: unset;
  -ms-flex-align: unset;
  align-items: unset;
  box-sizing: border-box;
  display: inline-block;
  -webkit-box-pack: unset;
  -webkit-justify-content: unset;
  -ms-flex-pack: unset;
  justify-content: unset;
  -webkit-order: unset;
  -ms-flex-order: unset;
  order: unset;
  padding: 0;
  vertical-align: top;
  width: 25%;
}

@media (max-width:992px) {
  .c0 {
    width: 25%;
  }
}

@media (max-width:768px) {
  .c0 {
    width: 25%;
  }
}

@media (max-width:576px) {
  .c0 {
    width: 25%;
  }
}

<div
  className="c0"
>
  This is a Grid component
</div>
`;

exports[`Grid component should render snapshot (with container) 1`] = `
.c0 {
  -webkit-align-content: unset;
  -ms-flex-line-pack: unset;
  align-content: unset;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: 'no-wrap';
  -ms-flex-wrap: 'no-wrap';
  flex-wrap: 'no-wrap';
  margin: 0 auto;
  max-width: 100%;
  width: 100%;
}

.c1 {
  -webkit-align-items: unset;
  -webkit-box-align: unset;
  -ms-flex-align: unset;
  align-items: unset;
  box-sizing: border-box;
  display: inline-block;
  -webkit-box-pack: unset;
  -webkit-justify-content: unset;
  -ms-flex-pack: unset;
  justify-content: unset;
  -webkit-order: unset;
  -ms-flex-order: unset;
  order: unset;
  padding: 0;
  vertical-align: top;
  width: 75%;
}

@media (max-width:992px) {
  .c1 {
    width: 75%;
  }
}

@media (max-width:768px) {
  .c1 {
    width: 75%;
  }
}

@media (max-width:576px) {
  .c1 {
    width: 75%;
  }
}

<div
  className="c0"
  size={100}
>
  <div
    className="c1"
  >
    This is a Grid component with container
  </div>
</div>
`;
